session-replay/Sources/DatadogSessionReplay/Processor/Privacy/TextObfuscator.swift

```diff
@@ -0,0 +1,32 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+internal struct TextObfuscator {
```

Comment:
I'm making it a separate type consciously (alternatively, it could be just a global function):
* By passing `TextObfuscator?` to wireframe builders we indicate if masking is necessary (although it could be optional closure too).
* If we have to, we can explore more options for improving the masking performance with implementing basic LRU cache (using `text.hashValue` as a key). For this (or other) perf tuning reasons, having an ability to manage internal state (in `class`) seems worthy.

Reply:
The caching would be a great optimisation indeed 👍  lets keep that in mind.

Reply:
👌 I created low-priority `RUMM-2671` to track perf optimisation ideas for text masking.